diff --git git/drivers/net/ethernet/smsc/smsc911x.c-orig git/drivers/net/ethernet/smsc/smsc911x.c
index 73be1cd..1860a4d 100644
--- git/drivers/net/ethernet/smsc/smsc911x.c-orig
+++ git/drivers/net/ethernet/smsc/smsc911x.c
@@ -529,9 +529,20 @@ static int smsc911x_mii_read(struct mii_bus *bus, int phyaddr, int regidx)
 
 	/* Confirm MII not busy */
 	if (unlikely(smsc911x_mac_read(pdata, MII_ACC) & MII_ACC_MII_BUSY_)) {
-		SMSC_WARN(pdata, hw, "MII is busy in smsc911x_mii_read???");
-		reg = -EIO;
-		goto out;
+		SMSC_WARN(pdata, hw, 
+			"MII is busy in smsc911x_mii_read???  phyaddr = %d  regidx = %d",
+			phyaddr, regidx);
+
+		for (i = 0; i < 100; i++) {
+			if (!(smsc911x_mac_read(pdata, MII_ACC) & MII_ACC_MII_BUSY_))
+				break;
+		}
+
+		if (i == 100) {
+			SMSC_WARN(pdata, hw, "Busy wait failed!!!");
+			reg = -EIO;
+			goto out;
+		}
 	}
 
 	/* Set the address, index & direction (read from PHY) */
