diff --git git/drivers/usb/otg/twl4030-usb.c-orig git/drivers/usb/otg/twl4030-usb.c
index c4a86da..4a34973 100644
--- git/drivers/usb/otg/twl4030-usb.c-orig
+++ git/drivers/usb/otg/twl4030-usb.c
@@ -143,6 +143,9 @@
 #define PMBR1				0x0D
 #define GPIO_USB_4PIN_ULPI_2430C	(3 << 0)
 
+#define TWL4030_USB_SEL_MADC_MCPC     (1<<3) 
+#define TWL4030_USB_CARKIT_ANA_CTRL   0xBB
+
 struct twl4030_usb {
 	struct usb_phy		phy;
 	struct device		*dev;
@@ -393,6 +396,11 @@ static void twl4030_phy_power(struct twl4030_usb *twl, int on)
 		regulator_disable(twl->usb1v8);
 		regulator_disable(twl->usb3v1);
 	}
+
+	/* Ensure that ADCIN3-6 are enabled as per TPS65950 manual */
+	twl4030_usb_write(twl, TWL4030_USB_CARKIT_ANA_CTRL,
+		twl4030_usb_read(twl, TWL4030_USB_CARKIT_ANA_CTRL) |
+		TWL4030_USB_SEL_MADC_MCPC);
 }
 
 static void twl4030_phy_suspend(struct twl4030_usb *twl, int controller_off)
